corehq/ex-submodules/soil/__init__.py

```diff
@@ -146,6 +146,8 @@ def task_id(self):
     @property
     def task(self):
         from soil.util import get_task
+        if not self.task_id:
+            return None
         return get_task(self.task_id)
```

Comment:
This pattern it makes it harder to reason about what you get when you access this property—the return type no longer has a consistent interface. I'm assuming from the code below that the result of this property is passed around to other places where it must now be tested to see if it's `None` or not. I also realize this is probably going to be harder to fix than what you can reasonably do quickly, so I'm not blocking on it.

Reply:
I agree with your point, but I think the LOE is too high to be worth fixing at this moment.  Thanks for pointing that out though - in particular I don't think I had thought about making sure a return type has a consistent interface before as clearly as you explained here.